From 81aa7501ce56cf7df7ee7be3bf6e7e37d1952dc8 Mon Sep 17 00:00:00 2001
From: CTCaer <ctcaer@gmail.com>
Date: Sun, 6 Mar 2022 04:27:54 +0000
Subject: [PATCH 04/39] libavformat: remove nvv4l2 probing mitigation

It was fixed properly with AV_CODEC_CAP_AVOID_PROBING flag.
---
 libavformat/demux.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/libavformat/demux.c b/libavformat/demux.c
index ae60a819d5..b19ab86d08 100644
--- a/libavformat/demux.c
+++ b/libavformat/demux.c
@@ -77,19 +77,6 @@ static const AVCodec *find_probe_decoder(AVFormatContext *s, const AVStream *st,
     if (codec_id == AV_CODEC_ID_H264)
         return avcodec_find_decoder_by_name("h264");
 #endif
-#if CONFIG_NVV4L2
-    /* NVV4L2 decoders depend on context init from base decoders */
-    if (codec_id == AV_CODEC_ID_HEVC)
-        return avcodec_find_decoder_by_name("hevc");
-    else if (codec_id == AV_CODEC_ID_MPEG2VIDEO)
-        return avcodec_find_decoder_by_name("mpeg2video");
-    else if (codec_id == AV_CODEC_ID_MPEG4)
-        return avcodec_find_decoder_by_name("mpeg4");
-    else if (codec_id == AV_CODEC_ID_VP8)
-        return avcodec_find_decoder_by_name("vp8");
-    else if (codec_id == AV_CODEC_ID_VP9)
-        return avcodec_find_decoder_by_name("vp9");
-#endif
 
     codec = ff_find_decoder(s, st, codec_id);
     if (!codec)
-- 
2.25.1

